### ISSUE TITLE
Feature that helps resetting password

### ISSUE THREAD

**Issue by RolandoV19** [MatrixEditor/hiktools#3]
I forgot the password for one of my cameras and have found that resetting password is a nightmare with hikvision, their later IP Cameras don't have a physical reset button any more. So you need to create a XML file using their SADP tool send it to their support email and then wait for them to send you back key to input that many times doesn't work. If more than 24hrs have passed from when you created the xml then the key has expired. 
I even have tried a firmware upgrade and then a version downgrade through TFTP and this did not clear the password either. 
It would be great to be able to locally reset the camera when needed

**Comment by MatrixEditor** [comment]
Have you tried the old Hikvision reset code ([docs](https://hiktools.readthedocs.io/en/latest/sadp.html#hiktools.sadp.hik_code))? It can be generated on several websites in the wild if you don't want to use this function.

**Comment by RolandoV19** [comment]
Hi

That reset code doesn’t work on newer firmwares, I have been looking through forums and on  GitHub.

The last firmware  that tool worked with was 5.5.3 or 5.5.4 , now they are on 5.5.89. Anyway I did trying and it doesn’t work.

My next step was was trying to find an old firmware to downgrade the camera to and then use that tool but looks like they are preventing that . I was trying to load 5.5.72 with a TFTP server from GitHub (https://github.com/wendelltron/hikvision-tftpd/tree/python3) and even when it said  it completed successfully it will boot into 5.5.89,  later i tried with the one form hikvision with same result. Them I tried with 5.5.82 and was accepted but nothing lower than that

So now stuck waiting on hikvision support.

Rgds

Sent from my iPhone

On Dec 7, 2022, at 21:54, MatrixEditor ***@***.***> wrote:

﻿

Have you tried the old Hikvision reset code (docs<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fhiktools.readthedocs.io%2Fen%2Flatest%2Fsadp.html%23hiktools.sadp.hik_code&data=05%7C01%7C%7C0b6434a5f1f643f6429f08dad830ac48%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638060000808360344%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=CnnP5Ud8VXIJGGPqW%2FHA5uqZgjoyvWL4kOSI3yiI89Y%3D&reserved=0>)? It can be generated on several websites in the wild if you don't want to use this function.
